prov/util/src/rocr_mem_monitor.c

```diff
@@ -275,9 +271,10 @@ static void rocr_mm_unsubscribe(struct ofi_mem_monitor *monitor,
 		if (!entry)
 			break;
 
-		cur_len -= MIN((uintptr_t) rocr_mm_entry_end_addr(entry) -
-			       (uintptr_t) cur_addr, cur_len);
-		cur_addr = rocr_mm_entry_end_addr(entry);
+		next_addr = (void *) ((uintptr_t) ofi_iov_end(&entry->iov) + 1);
+		cur_len -= MIN((uintptr_t) next_addr - (uintptr_t) cur_addr,
+			       cur_len);
+		cur_addr = next_addr;
```

Comment:
Should rocr_mm_entry_find() take cur_addr as input, instead of addr?

I think it would be more maintainable if the changes were moved to the end of the while loop, just so we don't adjust the loop control variables in the middle and expect the rest of the loop not to use them.

Reply:
> Should rocr_mm_entry_find() take cur_addr as input, instead of addr?

Yes. Thanks for catching this!

> I think it would be more maintainable if the changes were moved to the end of the while loop, just so we don't adjust the loop control variables in the middle and expect the rest of the loop not to use them.

Sounds good.